### ISSUE TITLE
Question: Log parameter type mismatch

### ISSUE THREAD

**Issue by LiorBanai** [meziantou/Meziantou.Analyzer#947]
I was going over the rules [MA0124](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0124.md) and [MA0125](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0125.md) and https://www.meziantou.net/roslyn-analyzer-to-check-the-types-of-structured-log-messages.htm and have few questions:

1. is there a way to validate all parameters of structured log messages of being same type? let say i use {Name} in many places and I would like to validate that the same type is used for it **but i don't care about that type itself**, only that there are no multiple types for the same parameter
2.  How can I specify custom types for MA0125  in `PropertyName;ExpectedType1;ExpectedType2;ExpectedType3` that are not CLR types? I'm getting:

 `The type 'MediaManager.Common.Types.PairingUuid' does not match any symbol of the compilation (https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0125.md)`

**Comment by meziantou** [comment]
> is there a way to validate all parameters of structured log messages of being same type? let say i use {Name} in many places and I would like to validate that the same type is used for it but i don't care about that type itself, only that there are no multiple types for the same parameter

No, but if you always have the same type, then it should be easy to add it to the configuration.

> How can I specify custom types for MA0125 in PropertyName;ExpectedType1;ExpectedType2;ExpectedType3 that are not CLR types? I'm getting:

You should be able to set any type that are available in the project. If you are unsure of the typename, you can use the attribute

```
// Requires the Meziantou.Analyzer.Annotations package
[assembly: Meziantou.Analyzer.Annotations.StructuredLogField("Count", typeof(PairingUuid))]
```

**Comment by LiorBanai** [comment]
"You should be able to set any type that are available in the project." 

The types reside in different project in the solution.

**Comment by meziantou** [comment]
Is the type accessible in your current project (ProjectReference)? I mean, does the following code works in your project?

```c#
PairingUuid dummy; // Or anything that actually use the type
```

**Comment by LiorBanai** [comment]
your comment helped to found the issue: I had global `LoggerParameterTypes.txt` file but a single project did not have reference to the project where the type was defined. After fixing it, it works.

Thanks!. Closing this.
